### PR TITLE
feat: add view tax component

### PR DESCRIPTION
**Pull request by OussEmaDevCode**
## Description
Added the view tax component, connected it via routing, designed its interface and connected it to the API.

## Related issues and discussion
#540 

## Screenshots, if any
Capture d’écran (232)

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by OussEmaDevCode** [comment]
@Jov03 @abhaychawla Hi, please have a look !

File: src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        <div fxFlex="50%" class="mat-body-strong">
+          Percentage
+        </div>
+
+        <div fxFlex="50%">
+          {{ taxComponentData.percentage }}
+        </div>
+
+        <div fxFlex="50%" class="mat-body-strong">
```
Comment (abhaychawla) [inline src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html]: Properties for debit account are missing, see the previous implementation here: https://github.com/openMF/community-app/blob/develop/app/views/products/tax/ViewTaxComponent.html
Update accordingly. Also, ensure you use ngIf (for optional properties) to only show those properties which have data.

**Comment by OussEmaDevCode** [comment]
@abhaychawla All required changes have been done !

File: src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html (hunk 1 above)
Comment (abhaychawla) [reply src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html]: Credit account data is also optional. Please check again add the condition for that as well.

File: src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          Percentage
+        </div>
+
+        <div fxFlex="50%">
+          {{ taxComponentData.percentage }}
+        </div>
+
+        <div *ngIf="taxComponentData.debitAccountType">
+          <div fxFlex="50%" class="mat-body-strong">
```
Comment (abhaychawla) [inline src/app/products/manage-tax-components/view-tax-component/view-tax-component.component.html]:
```suggestion
          <div fxFlex="50%" class="mat-body-strong" *ngIf="taxComponentData.debitAccountType">
```
Add the condition separately to the required inner divs, don't create a separate outer div. Make changes at all places accordingly.

**Comment by OussEmaDevCode** [comment]
@abhaychawla Done !

**Comment by OussEmaDevCode** [comment]
@abhaychawla Hi, can you please merge this commit (if it's good to merge) since I am currently working on the "Create the create tax component" task, and one of the required steps is to make sure that after tax creation we are redirected back to view tax component. Also since this commit contains major changes to the "products-routing-module" that are related to the manage tax components section, it will be easier to connect the new component via routing once this commit is merged.

**Comment by abhaychawla** [comment]
@OussEmaDevCode We need to look into the build failure issues that have been occurring frequently now, before we can merge. I understand your issue, but you will have to proceed as it is for now. The task description specifies that you can redirect to the list view if the view component does not exist at that point. So it won't be an issue.

**Comment by abhaychawla** [comment]
The build failure issue has been resolved! Its good to be merged now.

**Pull request merged**